agent/config/builder.go

```diff
@@ -106,14 +106,17 @@ func Load(opts LoadOpts) (LoadResult, error) {
 	if err := b.validate(cfg); err != nil {
 		return r, err
 	}
-	return LoadResult{RuntimeConfig: &cfg, Warnings: b.Warnings}, nil
+	watcherFiles := make([]string, len(opts.ConfigFiles))
+	copy(watcherFiles, opts.ConfigFiles)
```

Comment:
trivial:
you could move `structs.CloneStringSlice` to `lib/stringslice` as a new `Clone` method and then use that instead.